packages/discord.js/src/structures/interfaces/InteractionResponses.js

```diff
@@ -260,6 +260,22 @@ class InteractionResponses {
     this.replied = true;
   }
 
+  /**
+   * Responds to the interaction with an upgrade button.
+   * <warn>Only available for applications with monetization enabled.</warn>
+   * @returns {Promise<void>}
+   */
+  async sendPremiumUpgradeButton() {
```

Comment:
I concur, this name is not ideal, especially if Discord ever adds in more options. I'd just name this `sendPremiumRequired`